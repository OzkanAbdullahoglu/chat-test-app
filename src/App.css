.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

@font-face {
  font-family: 'BrownRegular';
  src: local('BrownRegular'), url(./assets/fonts/Brown-Regular.ttf) format('truetype');
}

body {
    color: rgba(0, 0, 0, 0.87);
    margin: 0;
    max-width: 480px;
    font-size: 0.875rem;
    font-family: "BrownRegular", "Helvetica", "Arial", sans-serif;
    font-weight: 400;
    line-height: 1.43;
    letter-spacing: 0.01071em;
    background-color: #fff;
}



.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
