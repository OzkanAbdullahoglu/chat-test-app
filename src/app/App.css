.App {
  text-align: center;
}

@font-face {
  font-family: 'BrownRegular';
  src: local('BrownRegular'), url(../assets/fonts/Brown-Regular.ttf) format('truetype');
}

body {
  color: rgba(0, 0, 0, 0.87);
  margin: 0;
  max-width: 480px;
  font-size: 0.875rem;
  font-family: "BrownRegular", "Helvetica", "Arial", sans-serif;
  font-weight: 400;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  background-color: #fff;
}
