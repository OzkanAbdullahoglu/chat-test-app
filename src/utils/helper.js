/* eslint-disable linebreak-style */
import React, { useState, useEffect } from 'react';
export const timeConverter = (timeStamp) => {
  if (!timeConverter.cache) {
    timeConverter.cache = {};
  }

  if (!timeConverter.cache[timeStamp]) {
    const timeStampMs = timeStamp * 1000;
    const dateObj = new Date(timeStampMs);
    const dayOfMonth = dateObj.getDate() < 10 ? `0${dateObj.getDate()}` : dateObj.getDate();
    const months = [
      'January', 'February', 'March', 'April', 'May', 'June',
      'July', 'August', 'September', 'October', 'November', 'December',
    ];
    const curMonth = months[dateObj.getMonth()];
    const curYear = dateObj.getFullYear();
    const curMinute = dateObj.getMinutes() < 10 ? `0${dateObj.getMinutes()}` : dateObj.getMinutes();
    const curHour = (dateObj.getHours() < 10 ? `0${dateObj.getHours()}` : dateObj.getHours());

    timeConverter.cache[timeStamp] = {
      date: `${curMonth} ${dayOfMonth},${curYear}`,
      time: `${curHour}:${curMinute}`,
    };
    return timeConverter.cache[timeStamp];
  }
  return timeConverter.cache[timeStamp];
};

export const isToday = (timestamp, date) =>{
  const today = Math.floor(Date.now() / 1000).toString();
  console.log(today);
  return timeConverter(today).date === timeConverter(timestamp).date ?
    'Today' : timeConverter(today).date === timeConverter(timestamp).date - 1 ?
    'Yesterday' : date;
};




function getCurrentViewDims() {
  const { innerWidth: width, innerHeight: height } = window;
  return {
    width,
    height,
  };
}

export default function useCurrentViewDims() {
  const [windowDimensions, setWindowDimensions] = useState(getCurrentViewDims());

  useEffect(() => {
    function handleResize() {
      setWindowDimensions(getCurrentViewDims());
    }

    window.addEventListener('scroll', handleResize);
    return () => window.removeEventListener('scroll', handleResize);
  }, []);

  return windowDimensions;
}

