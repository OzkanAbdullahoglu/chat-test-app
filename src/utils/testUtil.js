/* eslint-disable linebreak-style */
import { createStore, applyMiddleware } from 'redux';
import { middlewares } from '../store';
import rootReducer from '../reducers';

const chat = {
  requestedData: [
    {
      id: 1,
      direction: 'in',
      status: 'read',
      timestamp: '1577834102',
      text: 'Alice was beginning',
    },
  ],
  isPending: false,
  error: '',
  scrollDownEnable: true,
};

const message = {
  messagesEndRef: '',
};

const version = { number: '0.0.1' };

export const initialStates = { chat, message, version };

export const storeFactory = (initalState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
  return createStoreWithMiddleware(rootReducer, initalState);
};
