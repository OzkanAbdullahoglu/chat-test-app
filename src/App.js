/* eslint-disable linebreak-style */
import React, { useEffect } from 'react';
import { Provider, connect } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import { compose } from 'recompose';
import { persistor } from './store';
import './App.css';
import MessageList from './components/MessageList/MessageList';
import Header from './components/Header/Header';
import Spinner from './components/Spinner/Spinner';

import {
  chatActions,
  getPendingStatus,
} from '../src/reducers';

const App = ({ setRequestedData, isPending }) => {
  useEffect(() => { setRequestedData(); }, []);
  return (
    <PersistGate loading={null} persistor={persistor}>
      <Header incomingIcon={false} incomingAvatar={false} />
      <Header incomingIcon incomingAvatar />
      {isPending ? (
        <Spinner />
      ) : (
        <MessageList />
      )}

    </PersistGate>
  );
};

const mapStateToProps = (store) => ({
  isPending: getPendingStatus(store),
});

const withRedux = connect(
  mapStateToProps,
  { ...chatActions }
);

export default compose(withRedux)(App);
