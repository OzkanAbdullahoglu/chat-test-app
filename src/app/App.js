/* eslint-disable linebreak-style */
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import { compose } from 'recompose';
import PropTypes from 'prop-types';

import { persistor } from '../store';
import './App.css';
import MessageList from '../components/MessageList/MessageList';
import Header from '../components/Header/Header';
import Spinner from '../components/Spinner/Spinner';
import useCurrentViewDims from '../components/ViewPort';

import {
  chatActions,
  getPendingStatus,
} from '../reducers';

const App = ({ setRequestedData, isPending }) => {
  useEffect(() => { setRequestedData(); }, []);
  const { height, width } = useCurrentViewDims();
  return (
    <PersistGate loading={null} persistor={persistor}>
      <Header incomingIcon={false} incomingAvatar={false} />
      <Header incomingIcon incomingAvatar />
      {isPending ? (
        <Spinner data-test="spinner" />
      ) : (
        <MessageList
          width={width}
          height={height}
        />
      )}

    </PersistGate>
  );
};

App.propTypes = {
  setRequestedData: PropTypes.func,
  isPending: PropTypes.bool,
};

const mapStateToProps = (store) => ({
  isPending: getPendingStatus(store),
});

const withRedux = connect(
  mapStateToProps,
  { ...chatActions }
);

export default compose(withRedux)(App);
