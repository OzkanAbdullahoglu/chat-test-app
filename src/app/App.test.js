/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';
import { storeFactory } from '../utils/testUtil';

const defaultProps = { isPending: false };
const setup = () => {
  const store = storeFactory();
  return shallow(<App store={store} {...defaultProps} />).dive().dive();
};
describe('App', () => {
  const wrapper = setup();
  it('renders without crashing', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('contains two Header Component', () => {
    expect(wrapper.find('Connect(Header)').length).toEqual(2);
  });
  it('contains two MessageList Component', () => {
    expect(wrapper.find('Connect(MessageList)').length).toEqual(1);
  });
});

describe('App', () => {
  it('contains two Spinner Component when `isPending` prop is true', () => {
    const wrapper = setup();
    wrapper.setProps({ isPending: true });
    expect(wrapper.find('[data-test="spinner"]').length).toEqual(1);
  });
});
