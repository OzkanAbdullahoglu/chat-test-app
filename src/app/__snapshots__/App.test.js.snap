// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<PersistGate
  loading={null}
  persistor={
    Object {
      "dispatch": [Function],
      "flush": [Function],
      "getState": [Function],
      "pause": [Function],
      "persist": [Function],
      "purge": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Header)
    incomingAvatar={false}
    incomingIcon={false}
  />
  <Connect(Header)
    incomingAvatar={true}
    incomingIcon={true}
  />
  <Connect(MessageList)
    height={768}
    width={1024}
  />
</PersistGate>
`;
